pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentUtils.java

```diff
@@ -44,6 +44,16 @@ public static Integer getRealtimeSegmentPartitionId(String segmentName, String r
     if (llcSegmentName != null) {
       return llcSegmentName.getPartitionGroupId();
     }
+
+    try {
+      UploadedRealtimeSegmentName uploadedRealtimeSegmentName = new UploadedRealtimeSegmentName(segmentName);
```

Comment:
we may add a util method UploadedRealtimeSegmentName.of(segmentName) like LLCSegmentName.of()